File: spec/python/test_process_custom.py
import unittest

from process_custom import ProcessCustom

class MyCustomFx(object):
    def __init__(self, key, flag, some_array):
        self.key = key

    def decode(self, data):
        data

class TestProcessCustom(unittest.TestCase):
    def test_process_custom(self):
        r = ProcessCustom.from_file("src/process_rotate.bin")

        self.assertEqual(r.buf1, b"\x10\xb3\x94\x94\xf4")
        self.assertEqual(r.buf2, b"\x5f\xba\x7b\x93\x63\x23\x5f")
        self.assertEqual(r.buf3, b"\x29\x33\xb1\x38\xb1")

File: spec/lua/test_process_custom.lua
local luaunit = require("luaunit")

require("process_custom")

TestProcessCustom = {}

function TestProcessCustom:test_process_custom()
    local r = ProcessCustom:from_file("src/process_rotate.bin")

    luaunit.assertEquals(r.buf1, "\x10\xb3\x94\x94\xf4")
    luaunit.assertEquals(r.buf2, "\x5f\xba\x7b\x93\x63\x23\x5f")
    luaunit.assertEquals(r.buf3, "\x29\x33\xb1\x38\xb1")
end
